[2025-02-06T11:31:53.114+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-06T11:31:53.129+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: extract_news.extract_and_save_task scheduled__2025-02-05T00:00:00+00:00 [queued]>
[2025-02-06T11:31:53.137+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: extract_news.extract_and_save_task scheduled__2025-02-05T00:00:00+00:00 [queued]>
[2025-02-06T11:31:53.138+0000] {taskinstance.py:2866} INFO - Starting attempt 2 of 2
[2025-02-06T11:31:53.150+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): extract_and_save_task> on 2025-02-05 00:00:00+00:00
[2025-02-06T11:31:53.155+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=73) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-02-06T11:31:53.157+0000] {standard_task_runner.py:72} INFO - Started process 75 to run task
[2025-02-06T11:31:53.158+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'extract_news', 'extract_and_save_task', 'scheduled__2025-02-05T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/extract_dag.py', '--cfg-path', '/tmp/tmpoctw5ktz']
[2025-02-06T11:31:53.159+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask extract_and_save_task
[2025-02-06T11:31:53.205+0000] {task_command.py:467} INFO - Running <TaskInstance: extract_news.extract_and_save_task scheduled__2025-02-05T00:00:00+00:00 [running]> on host aeb3e4fd1105
[2025-02-06T11:31:53.469+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='extract_news' AIRFLOW_CTX_TASK_ID='extract_and_save_task' AIRFLOW_CTX_EXECUTION_DATE='2025-02-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-02-05T00:00:00+00:00'
[2025-02-06T11:31:53.470+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-02-06T11:31:53.470+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-02-06T11:31:53.471+0000] {logging_mixin.py:190} INFO - Current task name:extract_and_save_task state:running start_date:2025-02-06 11:31:53.130442+00:00
[2025-02-06T11:31:53.471+0000] {logging_mixin.py:190} INFO - Dag name:extract_news and current dag run status:running
[2025-02-06T11:31:53.472+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-06T11:31:53.473+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/extract.py", line 19, in extract_and_save
    reddit = praw.Reddit(
             ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/praw/util/deprecate_args.py", line 46, in wrapped
    return func(**dict(zip(_old_args, args)), **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/praw/reddit.py", line 259, in __init__
    raise MissingRequiredAttributeException(
praw.exceptions.MissingRequiredAttributeException: Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.
[2025-02-06T11:31:53.485+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-02-06T11:31:53.486+0000] {logging_mixin.py:190} INFO - Task start:2025-02-06 11:31:53.130442+00:00 end:2025-02-06 11:31:53.485340+00:00 duration:0.354898
[2025-02-06T11:31:53.486+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): extract_and_save_task> dag:<DAG: extract_news> dagrun:<DagRun extract_news @ 2025-02-05 00:00:00+00:00: scheduled__2025-02-05T00:00:00+00:00, state:running, queued_at: 2025-02-06 11:26:46.229033+00:00. externally triggered: False>
[2025-02-06T11:31:53.487+0000] {logging_mixin.py:190} INFO - Failure caused by Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.
[2025-02-06T11:31:53.487+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=extract_news, task_id=extract_and_save_task, run_id=scheduled__2025-02-05T00:00:00+00:00, execution_date=20250205T000000, start_date=20250206T113153, end_date=20250206T113153
[2025-02-06T11:31:53.498+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-06T11:31:53.499+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 8 for task extract_and_save_task (Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.; 75)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/extract.py", line 19, in extract_and_save
    reddit = praw.Reddit(
             ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/praw/util/deprecate_args.py", line 46, in wrapped
    return func(**dict(zip(_old_args, args)), **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/praw/reddit.py", line 259, in __init__
    raise MissingRequiredAttributeException(
praw.exceptions.MissingRequiredAttributeException: Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.
[2025-02-06T11:31:53.534+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-02-06T11:31:53.550+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-06T11:31:53.553+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
